Add promise based API to RTCPeerConnection

dbrgn opened the pull request:
This is an improvement to existing type definition. Source: https://www.w3.org/TR/webrtc/

- Added new promise based API to `RTCPeerConnection`
- The old API is kept for legacy purposes (https://www.w3.org/TR/webrtc/#h-legacy-interface-extensions)

So far, it has not yet been reviewed by a DefinitelyTyped member. Could one (or two) of @smithkl42 @nestalk @akonwi @iislucas @nakakura @martinduparc @Garciat @groege @lgrahl maybe review?

iislucas commented on webrtc/RTCPeerConnection-tests.ts:
```diff
@@ -1,6 +1,8 @@
 /// <reference path="MediaStream.d.ts" />
 /// <reference path="RTCPeerConnection.d.ts" />
 
+let voidpromise: Promise<void>;
```
I don't think this abbreviation helps much; it's not really much more concise, and means that is someone jumps into the middle of the file, they have a to search and find this definition. I'd recommend just using the `Promise<void>` below instead.

iislucas commented:
:+1: Nice! Great to see the new promise syntax of WebRTC being supported. :)

I notice you're removing the tests for the old syntax, which I think is ok. But you could also have left them in and just added new ones for the new syntax/calls you added.

dbrgn commented:
> Great to see the new promise syntax of WebRTC being supported. :)

Definitely :)

> I notice you're removing the tests for the old syntax, which I think is ok. But you could also have left them in and just added new ones for the new syntax/calls you added. 

That was by mistake. First I replaced all calls, until I noticed that they're being kept for legacy reasons. Then I added the code back, but forgot about the tests.

I don't think it's that important though. They're deprecated anyways :)

vvakame commented:
@smithkl42 can I merge this PR?

mhegazy commented:
No response from other authors for over 2 weeks. Merging.
